src/getinfo.sml

```diff
@@ -0,0 +1,242 @@
+                                    if isSmallerThan span span'
+                                    then 
+                                        (case printGoodPart env item span of
+                                             NONE => SOME prev
+                                          | SOME desc => SOME {desc = desc, span = span, env = env, item = item})
+                                    else SOME prev
+                        in
+                            {smallest = smallest, smallestgoodpart = smallestgoodpart}
+                        end
+
+        (* Look for item at input position *)
+        (* We're looking for two things simultaneously: *)
+        (* 1. The "smallest" part, ie. the one of which the source span is the smallest *)
+        (* 2. The "smallestgoodpart" part, ie. the one of which the source span is the smallest AND has a special case in printGoodPart *)
+        (* If we end up with a smallestgoodpart, we'll show that one since that one is probably more useful  *)
+        (* TODO source spans of XML and SQL sources are weird and you end *)
+        (*   up with eg: a span from eg 1-5 and another from 2-6, makes no sense? *)
+        (*   That's one of the reasons why we're searching for the two things mentioned above *)
+        val result =
+            U.Decl.foldB
+                { kind = fn (env, (k, span), acc) => add env (Kind (k, span)) span acc,
+                  con = fn (env, (k, span), acc) => add env (Con (k, span)) span acc,
+                  exp = fn (env, (k, span), acc) => add env (Exp (k, span)) span acc,
+                  sgn_item = fn (env, (k, span), acc) => add env (Sgn_item (k, span)) span acc,
+                  sgn = fn (env, (k, span), acc) => add env (Sgn (k, span)) span acc,
+                  str = fn (env, (k, span), acc) => add env (Str (k, span)) span acc,
+                  decl = fn (env, (k, span), acc) => add env (Decl (k, span)) span acc,
+                  bind = fn (env, binder) =>
+                            case binder of
+                                U.Decl.RelK x => E.pushKRel env x
+                              | U.Decl.RelC (x, k) => E.pushCRel env x k
+                              | U.Decl.NamedC (x, n, k, co) => E.pushCNamedAs env x n k co
+                              | U.Decl.RelE (x, c) => E.pushERel env x c
+                              | U.Decl.NamedE (x, c) => #1 (E.pushENamed env x c)
+                              | U.Decl.Str (x, n, sgn) => #1 (E.pushStrNamed env x sgn)
+                              | U.Decl.Sgn (x, n, sgn) => #1 (E.pushSgnNamed env x sgn)
+                }
+                env
+                { smallestgoodpart = NONE
+                , smallest = { item = Str (str, { file = fileName
+                                                , first = { line = 0, char = 0}
+                                                , last = { line = 99999, char = 0} })
```

Comment:
I think it would be nicer to replace these 99999s with [`Int.maxInt`](http://sml-family.org/Basis/integer.html), to make clear that we are just choosing the largest possible value, which we expect to be replaced quickly by more specific bounds.

Reply:
OK, done. It's a bit stupid that Int.maxInt is an option, but OK... :P

Reply:
Yeah, you could call it a bit of laziness on the parts of the library designers.  They wanted one signature for all integer-type modules, and there are unbounded integer types out there.